packages/java/tests/spring/react-signals/src/test/java/com/vaadin/hilla/test/SignalBasicSecurityIT.java

```diff
@@ -0,0 +1,162 @@
+package com.vaadin.hilla.test;
+
+import com.vaadin.flow.component.button.testbench.ButtonElement;
+import com.vaadin.flow.component.textfield.testbench.PasswordFieldElement;
+import com.vaadin.flow.component.textfield.testbench.TextFieldElement;
+import com.vaadin.flow.testutil.ChromeBrowserTest;
+import com.vaadin.testbench.parallel.Browser;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.BlockJUnit4ClassRunner;
+import org.openqa.selenium.By;
+
+@RunWith(BlockJUnit4ClassRunner.class)
+public class SignalBasicSecurityIT extends ChromeBrowserTest {
+
+    @Override
+    @Before
+    public void setup() throws Exception {
+        setDesiredCapabilities(Browser.CHROME.getDesiredCapabilities());
+        super.setup();
+        getDriver().get(getRootURL() + "/SharedSignalSecurity");
+        waitForPageToLoad();
+        resetCountersAndValidateInitialValues();
+    }
+
+    private void waitForPageToLoad() {
+        waitForElementPresent(By.ById.id("userCounter"));
+        waitForElementPresent(By.ById.id("adminCounter"));
+        waitUntil(driver -> $("span").id("UserCounterFromServer")
+                .getText() != null);
+        waitUntil(driver -> $("span").id("AdminCounterFromServer")
+                .getText() != null);
+    }
+
+    private void resetCountersAndValidateInitialValues() {
+        clickButton("reset");
+        waitFor(1000);
+        Assert.assertEquals(20, fetchCounterValue("User"));
+        Assert.assertEquals(30, fetchCounterValue("Admin"));
+    }
+
+    @Test
+    public void anonymousAccessIsNotAvailableToUserAndAdminCounterSignals() {
+        // check that the counters are not accessible for anonymous users:
+        Assert.assertEquals(-1, getCounterSignalValue("userCounter"));
+        Assert.assertEquals(-1, getCounterSignalValue("adminCounter"));
+    }
+
+    @Test
+    public void userCounterSignalIsAccessibleForLoggedInUser() {
+
+        loginAs("user");
+
+        Assert.assertEquals(20, fetchCounterValue("User")); // shows server and
+                                                            // related endpoint
+                                                            // are working
+        Assert.assertEquals(20, getCounterSignalValue("userCounter")); // user
+                                                                       // counter
+                                                                       // should
+                                                                       // be
+                                                                       // accessible
```

Comment:
Seems like the end-line comments aren't playing nicely with autoformatter:

```suggestion
        // Shows server and related endpoint are working:
        Assert.assertEquals(20, fetchCounterValue("User"));

        // User counter should be accessible:
        Assert.assertEquals(20, getCounterSignalValue("userCounter")); 
```

Reply:
Also below in the same file.

Reply:
To avoid extra builds and re-review requests, I will merge this as it is, and will fix it on the next PR.